lib/features/modeling/ElementFactory.js

```diff
@@ -211,7 +211,7 @@ ElementFactory.prototype.createElement = function(elementType, attrs) {
   }
 
   if (is(businessObject, 'bpmn:ExclusiveGateway')) {
-    di.isMarkerVisible = true;
+    attrs = applyAttribute(di, attrs, 'isMarkerVisible');
```

Comment:
Let's set the attribute to a non-true value only if it's set explicitly (either to `undefined` or `false`) which can be checked via `attrs.hasOwnProperty('isMarkerVisible')`. This way we do not introduce a breaking change.

Reply:
I can only _set_ `undefined` by excluding the attribute from the XML. As a result, we would incorrectly show the marker if the attribute is missing, but still hide it if the attribute were `false`.

Just to be clear, is that the intention?

Reply:
@barmac Thanks for the explanation.

Maybe I confused myself over this, but I fail to see how I can pass `isMarkerVisible: undefined`.

The property doesn't become available when using either the following code or trying the same from within the ContextPadProvider.

`elementFactory.createShape({ type: 'bpmn:ExclusiveGateway' });`
`elementFactory.createShape({ type: 'bpmn:ExclusiveGateway', isMarkerVisible: undefined });`
`elementFactory.createShape({ type: 'bpmn:ExclusiveGateway' }, { isMarkerVisible: undefined });`

So I don't have a chance to use `hasOwnProperty` there. Thus, I cannot differ `isMarkerVisible: undefined` from missing property.

Reply:
Never mind. I learned something about JavaScript and developer tools. Thanks.